spanner/transaction_test.go

```diff
@@ -1342,6 +1342,191 @@ func TestPriorityInQueryOptions(t *testing.T) {
 	}
 }
 
+func TestLastStatement_Update(t *testing.T) {
+	t.Parallel()
+
+	server, client, teardown := setupMockedTestServer(t)
+	defer teardown()
+
+	if _, err := client.ReadWriteTransaction(context.Background(), func(ctx context.Context, tx *ReadWriteTransaction) error {
+		if _, err := tx.Update(context.Background(), NewStatement(UpdateBarSetFoo)); err != nil {
+			return err
+		}
+		if _, err := tx.UpdateWithOptions(context.Background(), NewStatement(UpdateBarSetFoo), QueryOptions{LastStatement: true}); err != nil {
+			return err
+		}
+		return nil
+	}); err != nil {
+		t.Fatalf("transaction failed: %v", err)
+	}
+	requests := drainRequestsFromServer(server.TestSpanner)
+	executeRequests := requestsOfType(requests, reflect.TypeOf(&sppb.ExecuteSqlRequest{}))
+	if g, w := len(executeRequests), 2; g != w {
+		t.Fatalf("num requests mismatch\n Got: %v\nWant: %v", g, w)
+	}
+	for i := 0; i < len(executeRequests); i++ {
+		if g, w := executeRequests[i].(*sppb.ExecuteSqlRequest).LastStatement, i == len(executeRequests)-1; g != w {
+			t.Fatalf("%d: last statement mismatch\n Got: %v\nWant: %v", i, g, w)
+		}
+	}
```

Comment:
This test shows how to use the `LastStatement` option in the client library. Setting the `LastStatement: true` option causes the client library to include the option in the request, which again allows Spanner to optimize the transaction execution.